pkg/status/combinedstatus-resolver.go

```diff
@@ -0,0 +1,307 @@
+/*
+Copyright 2024 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package status
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/kubestellar/kubestellar/api/control/v1alpha1"
+	"github.com/kubestellar/kubestellar/pkg/abstract"
+	"github.com/kubestellar/kubestellar/pkg/binding"
+	"github.com/kubestellar/kubestellar/pkg/util"
+)
+
+type CombinedStatusResolver interface {
+	// TODO: API for generating/comparing combinedstatus resolutions
+
+	// NoteBindingResolution notes a binding resolution for status collection.
+	//
+	// 1. If `deleted` is true, the associated combinedstatus resolutions are
+	// removed from memory. The same is done if a resolution no longer requires
+	// status collection.
+	//
+	// 2. Excessive combinedstatus resolutions are removed if they are no longer
+	// associated with the binding.
+	//
+	// 3. For every workload object associated with one or more statuscollectors, a
+	// combinedstatus resolution is created/updated.
+	// In the case of creation, the binding.Resolution's OwnerReference is used
+	// to identify the bindingpolicy that the resolution is associated with.
+	//
+	// The returned array contains:
```

Comment:
It's not an array.

Since it contains two sorts of things, maybe return two homogenous sets?